pilot/pkg/bootstrap/server.go

```diff
@@ -647,29 +586,42 @@ func (s *Server) initMCPConfigController(args *PilotArgs) error {
 		msgSizeOption := grpc.WithDefaultCallOptions(grpc.MaxCallRecvMsgSize(args.MCPMaxMessageSize))
 
 		conn, err := grpc.DialContext(
-			ctx, configSource.Address,
-			securityOption, msgSizeOption, keepaliveOption, initialWindowSizeOption, initialConnWindowSizeOption)
+			ctx,
+			configSource.Address,
+			securityOption,
+			msgSizeOption,
+			keepaliveOption,
+			initialWindowSizeOption,
+			initialConnWindowSizeOption)
 		if err != nil {
 			log.Errorf("Unable to dial MCP Server %q: %v", configSource.Address, err)
 			cancel()
 			return err
 		}
-
-		mcpController := coredatamodel.NewController(options)
-		sinkOptions := &sink.Options{
-			CollectionOptions: collections,
-			Updater:           mcpController,
-			ID:                clientNodeID,
-			Reporter:          reporter,
-		}
-
-		cl := mcpapi.NewResourceSourceClient(conn)
-		mcpClient := sink.NewClient(cl, sinkOptions)
-		configz.Register(mcpClient)
-		clients = append(clients, mcpClient)
-
 		conns = append(conns, conn)
-		configStores = append(configStores, mcpController)
+		s.mcpController(args, conn, reporter, &clients, &configStores)
+
+		// create MCP SyntheticServiceEntryController
+		if resourceContains(configSource.SubscribedResources, meshconfig.Resource_SERVICE_REGISTRY) {
+
+			//TODO(Nino-K): https://github.com/istio/istio/issues/16976
+			args.Service.Registries = []string{string(serviceregistry.MCPRegistry)}
```

Comment:
If https://github.com/istio/istio/pull/12276/files#r334767650 is true, why set `MCP` redundantly?